### ISSUE TITLE
Add more predictors to make predictions more stable

### ISSUE THREAD

**Issue by NIlS-Z99** [NIlS-Z99/Invest-Tool#2]
Add "Decision Tree" predictors of different month window sizes and merge all predictors with a "Random Forest" predictor

Code Snippets:
`import pandas as pd
import numpy as np
import datetime as dt
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

# Collect historical stock data
stock_data = pd.read_csv('historical_stock_data.csv')

# Preprocess the data
stock_data.drop_duplicates(inplace=True)
stock_data.dropna(inplace=True)
stock_data['Date'] = pd.to_datetime(stock_data['Date'])
stock_data.set_index('Date', inplace=True)

# Feature engineering
stock_data['Moving Average'] = stock_data['Close'].rolling(window=20).mean()
stock_data['RSI'] = RSI(stock_data['Close'], window_length=14)
stock_data['MACD'] = MACD(stock_data['Close'], fast_window=12, slow_window=26)

# Train-test split
train_size = int(len(stock_data) * 0.8)
train_data, test_data = stock_data[:train_size], stock_data[train_size:]

# Train model
rf_model = RandomForestRegressor(n_estimators=100, random_state=0)
X_train, y_train = train_data.drop(['Close'], axis=1), train_data['Close']
rf_model.fit(X_train, y_train)

# Test model
X_test, y_test = test_data.drop(['Close'], axis=1), test_data['Close']
y_pred = rf_model.predict(X_test)

# Model evaluation
mae = mean_absolute_error(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

# Make predictions for the next year
last_date = stock_data.index[-1]
next_year = last_date + dt.timedelta(days=365)
next_year_data = stock_data.loc[last_date:next_year]
next_year_predictions = rf_model.predict(next_year_data.drop(['Close'], axis=1))

# Output predictions
print("Predictions for the next year:")
print(next_year_predictions)`

`import pandas as pd
import numpy as np
import datetime as dt
from sklearn.tree import DecisionTreeRegressor
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

# Collect historical stock data
stock_data = pd.read_csv('historical_stock_data.csv')

# Preprocess the data
stock_data.drop_duplicates(inplace=True)
stock_data.dropna(inplace=True)
stock_data['Date'] = pd.to_datetime(stock_data['Date'])
stock_data.set_index('Date', inplace=True)

# Feature engineering
stock_data['Moving Average'] = stock_data['Close'].rolling(window=20).mean()
stock_data['RSI'] = RSI(stock_data['Close'], window_length=14)
stock_data['MACD'] = MACD(stock_data['Close'], fast_window=12, slow_window=26)

# Train-test split
train_size = int(len(stock_data) * 0.8)
train_data, test_data = stock_data[:train_size], stock_data[train_size:]

# Train model
dt_model = DecisionTreeRegressor(random_state=0)
X_train, y_train = train_data.drop(['Close'], axis=1), train_data['Close']
dt_model.fit(X_train, y_train)

# Test model
X_test, y_test = test_data.drop(['Close'], axis=1), test_data['Close']
y_pred = dt_model.predict(X_test)

# Model evaluation
mae = mean_absolute_error(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

# Make predictions for the next year
last_date = stock_data.index[-1]
next_year = last_date + dt.timedelta(days=365)
next_year_data = stock_data.loc[last_date:next_year]
next_year_predictions = dt_model.predict(next_year_data.drop(['Close'], axis=1))

# Output predictions
print("Predictions for the next year:")
print(next_year_predictions)`
